File: packages/core/src/config/analytics.ts
// Analytics・Benchmark（merged の `analytics` + `benchmark`、どちらも optional）。
//
// `benchmark.channels` は JSON 構造を verbatim 保持する（snake key を camel 化しない）。

import { z } from "zod";

/** `analytics` + `benchmark` の合成（どちらも optional）。 */
export const Analytics = z
  .object({
    analytics: z
      .object({
        collection_filter_keywords: z.array(z.string()).default([]),
      })
      .strict()
      .prefault({}),
    benchmark: z
      .object({
        channels: z.array(z.record(z.string(), z.unknown())).default([]),
      })
      .strict()
      .prefault({}),
  })
  .transform((o) => ({
    benchmark: { channels: o.benchmark.channels },
    collectionFilterKeywords: o.analytics.collection_filter_keywords,
  }));

export type Analytics = z.infer<typeof Analytics>;

File: src/youtube_automation/configuration/analytics.py
"""Analytics・Benchmark の責務別 dataclass."""

from __future__ import annotations

from dataclasses import dataclass, field


@dataclass(frozen=True)
class Benchmark:
    """`benchmark` セクション（optional）."""

    channels: list[dict] = field(default_factory=list)


@dataclass(frozen=True)
class Analytics:
    """`analytics` セクション + `benchmark` セクションの合成（どちらも optional）."""

    collection_filter_keywords: list[str] = field(default_factory=list)
    benchmark: Benchmark = field(default_factory=Benchmark)
